localstack/services/events/provider.py

```diff
@@ -622,16 +616,27 @@ def process_event_with_input_transformer(input_transformer: Dict, event: Dict) -
     return templated_event
 
 
-def process_events(event: Dict, targets: list[Dict]):
```

Comment:
The diff, especially around here, is a little messy-looking but it's mainly creating class `PutEventsHandler` that is callable, the `__call__` method being the same implement as `events_handler_put_events`. This class also absorbs some functions into internal methods, mainly because they're only used and needed for the handler's sake. Implementation details should remain relatively the same, save for the executor-publisher being involved instead of directly calling `send_event_to_target`